### ISSUE TITLE
Grid + Put: cannot "put" windows after placing in "top-left grid" (corner), etc.

### ISSUE THREAD

**Issue by jessexknight** [compiz-reloaded/compiz#111]
I have two monitors, and use _Put_ to move windows between them, and _Grid_ to place them within each screen. The problem:

For any windows in the "quarter-screen" state from _Grid_ (Top Left, Top Right, Bottom Left, Bottom Right), when I attempt to _Put_ the window to the other screen...

**Expected behaviour:**
The _Put_ move places the window in the same top-left etc. position on the new screen.

**Actual behaviour:**
The _Put_ move is rejected, and the window pops back to where it was originally. 

**Possibly related:**
When _Put_ moving "half-screen" windows, _some_ of the coordinates seem to be "checked" after the move, to match the new monitor:
For Bottom / Top windows (wide), the width is updated to match the new monitor, but not the height.
For Left / Right windows (tall), the height is updated to match the new monitor, but not the width.

**My hypothesis:**
The [_Put_ module](https://github.com/compiz-reloaded/compiz-plugins-main/blob/master/src/put/put.c) does not consider different monitor sizes correctly when computing the final coordinates. Specifically, the LOCs in the module around 400+.

... unfortunately I don't have 2 same-size monitors to test the relevance of different monitors.

**Comment by soreau** [comment]
@jessexknight Hi, when you say 'put move', to which binding are you referring specifically? I briefly tested 'Put to Next Output' binding and it seemed to work regardless if the window had been gridded to a corner, on dual same-res outputs.

**Comment by jessexknight** [comment]
@soreau Thanks for the quick reply.

**Binding:**
Yes, I am using 'Put to Next Output', with keyboard binding. I've tried different key bindings, same issue.
**Same resolution test:**
I put both monitors at the same resolution, and still had the same issue. So, maybe it is a conflicting setting I'm using... Any ideas?

**Comment by soreau** [comment]
Is the window somehow losing focus after gridding? In other words, if you click on the window titlebar after grid-to-corner action, does put move work? Does this only happen on a specific window or does it happen all the time? If you manually resize and move the window into a corner like grid would place it, does put move work then?

**Comment by jessexknight** [comment]
1. _Is the window somehow losing focus after gridding? In other words, if you click on the window titlebar after grid-to-corner action, does put move work?_
The window is still in focus after gridding. If I click the title bar after grid-to-corner action, the put move still does not work.

2. _Does this only happen on a specific window or does it happen all the time?_
It happens all the time.

3. _If you manually resize and move the window into a corner like grid would place it, does put move work then?_
Bingo! After grid-to-corner, if I resize the window in any way, the put-move **works**. This includes just clicking the edge (as if to resize, but not dragging at all). This seems to imply windows have a 'corner' state assigned by grid, which prevents the put-move, and that, this state is destroyed as soon as the window is resized another way.

**Comment by soreau** [comment]
That's a good find to fixing the bug but since I can't reproduce the problem, I can't be of much help other than to say that debugging would probably start with printf statements in the put plugin to try and find out why it's failing in this case.

**Comment by jessexknight** [comment]
Thanks.
Re. you debugging: you _could_ change one of your two same monitors to a different resolution.
Re. me debugging: any tips on compiling & debugging compiz safely? I haven't touched C in a while...

**Comment by soreau** [comment]
Yes, I did try monitors with different resolutions, it still worked here. To build the put plugin, you would have to clone, build and install compiz-plugins-main using something like `git clone https://github.com/compiz-reloaded/compiz-plugins-main && cd compiz-plugins-main && ./autogen.sh --prefix=/usr && make && sudo make install` then you can make changes to src/put/put.c (in the plugins-main dir) and make && sudo make install, then restart compiz with `compiz --replace` (from a terminal to see the output)

**Comment by jessexknight** [comment]
**Re. reproducing the issue**
Interesting that you still cannot reproduce the issue... I just remembered another quirk of my setup possibly worth mentioning: one of my monitors is VGA, while the other is DVI. Do you think this is significant?

**Re. building src for debugging**
Thanks for the info. Here's hoping I don't break anything :}

**Comment by jessexknight** [comment]
Trying to build from source, I get:
```
$ ./autogen.sh --prefix=/usr

autoreconf: Entering directory `.'
autoreconf: running: autopoint --force
autoreconf: running: aclocal --force -I m4 ${ACLOCAL_FLAGS}
autoreconf: configure.ac: tracing
autoreconf: configure.ac: not using Libtool
autoreconf: running: /usr/bin/autoconf --force
autoreconf: running: /usr/bin/autoheader --force
autoreconf: running: automake --add-missing --copy --force-missing
src/animation/Makefile.am:30: error: Libtool library used but 'LIBTOOL' is undefined
src/animation/Makefile.am:30:   The usual way to define 'LIBTOOL' is to add 'LT_INIT'
src/animation/Makefile.am:30:   to 'configure.ac' and run 'aclocal' and 'autoconf' again.
src/animation/Makefile.am:30:   If 'LT_INIT' is in 'configure.ac', make sure
src/animation/Makefile.am:30:   its definition is in aclocal's search path.
src/colorfilter/Makefile.am:26: error: Libtool library used but 'LIBTOOL' is undefined
src/colorfilter/Makefile.am:26:   The usual way to define 'LIBTOOL' is to add 'LT_INIT'
src/colorfilter/Makefile.am:26:   to 'configure.ac' and run 'aclocal' and 'autoconf' again.
src/colorfilter/Makefile.am:26:   If 'LT_INIT' is in 'configure.ac', make sure
src/colorfilter/Makefile.am:26:   its definition is in aclocal's search path.
... [21 more similar errors] ...
autoreconf: automake failed with exit status: 1
```

From this [StackExchange post](https://stackoverflow.com/questions/22526375/where-is-aclocals-search-path), I'm not sure which symbolic link to make in this case, if any.

**Comment by soreau** [comment]
@jessexknight I think you need to install libtool.
